src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -168,23 +168,27 @@ protected override void AddEmbeddedResourcesFromAddedModules(ArrayBuilder<Cci.Ma
 
         internal override SynthesizedAttributeData SynthesizeEmbeddedAttribute()
         {
-            if ((object)_lazyEmbeddedAttribute != null)
-            {
-                return new SynthesizedAttributeData(
-                    _lazyEmbeddedAttribute.Constructor,
-                    ImmutableArray<TypedConstant>.Empty,
-                    ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
-            }
+            return new SynthesizedAttributeData(
+                _lazyEmbeddedAttribute.Constructors[0],
+                ImmutableArray<TypedConstant>.Empty,
+                ImmutableArray<KeyValuePair<string, TypedConstant>>.Empty);
+        }
 
-            return base.SynthesizeEmbeddedAttribute();
+        protected override SynthesizedAttributeData SynthesizeNullableAttribute(WellKnownMember member, ImmutableArray<TypedConstant> arguments = default)
+        {
+            var constructorIndex = (member == WellKnownMember.System_Runtime_CompilerServices_NullableAttribute__ctorTransformFlags) ? 1 : 0;
+            return new SynthesizedAttributeData(
+                _lazyNullableAttribute.Constructors[constructorIndex],
```

Comment:
>_lazyNullableAttribute [](start = 16, length = 22)

It looks like this code is going to cause a NullReferenceException in case when the attribute doesn't have to be embedded. In this case we should fall back to the regular way of synthesizing attributes by using already available type.

Reply:
I will handle existing attribute types in a separate PR. See `ExplicitAttributeFromSource` test.

Reply:
>  I will handle existing attribute types in a separate PR.

Why? It feels like we should simply follow the pattern of the other similar methods, i.e. call the base when ```_lazyNullableAttribute``` is null. Is there something special about Nullable attribute that I am missing. Is there a reason to deviate?

Reply:
More tests will be required when supporting existing `NullableAttribute` definitions, testing for missing constructors, etc.